### PR TITLE
Add getOrElse method

### PR DESCRIPTION
**Pull request by x0ji0d**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,6 +20,8 @@ trait Cache[F[_], K, V] {
 
   def get(key: K): F[Option[V]]
 
+  def getOrElse(key: K, default: => V): F[V]
```
Comment (t3hnar) [inline README.md]: why not `def getOrElse(key: K, default: => F[V]): F[V]` ?

Comment (x0ji0d) [reply README.md]: Good question :) will do

File: src/main/scala/com/evolutiongaming/scache/CacheFenced.scala [hunk 2]
```diff
@@ -29,12 +29,14 @@ object CacheFenced {
...
   }
 
-  def apply[F[_] : FlatMap, K, V](cache: Cache[F, K, V], fence: F[Unit]): Cache[F, K, V] = {
+  def apply[F[_] : Monad, K, V](cache: Cache[F, K, V], fence: F[Unit]): Cache[F, K, V] = {
 
     new Cache[F, K, V] {
 
       def get(key: K) = cache.get(key)
 
+      def getOrElse(key: K, default: => F[V]): F[V] = get(key).flatMap(_.fold(default)(_.pure[F]))
```
Comment (t3hnar) [inline src/main/scala/com/evolutiongaming/scache/CacheFenced.scala]: going through changes, I can see that basically there is only one implementation available, should we maybe just add that as extension method via implicit class to ?

Comment (x0ji0d) [reply src/main/scala/com/evolutiongaming/scache/CacheFenced.scala]: Put to the traits ¯\_(ツ)_/¯

**Pull request merged**